dvc/remote/ssh/connection.py

```diff
@@ -143,7 +144,16 @@ def walk_files(self, directory):
 
     def remove(self, path):
         self._sftp_connect()
-        self._sftp.remove(path)
+        try:
+            self._sftp.remove(path)
+        except FileNotFoundError as exc:
+            if exc.errno == errno.ENOENT:
+                logger.warning(
```

Comment:
Also, we should be using the same operation semantics for all remotes, I think. Can we confirm that all of them do not throw if path does not exist? Alternative to this is to actually ensure this semantic on the caller side. Wrap FileNotFoundError (IoError) here into DvcNotFoundException and catch upstream. Your thoughts, @efiop @mroutis ?

Reply:
@shcheklein This approach is fine, this is how we handle it in local remote too. Other remotes don't have the same behaviour though, we'll need to fix that in the future.

Reply:
Same here https://github.com/iterative/dvc/pull/1737#discussion_r266248094 .